### DISCUSSION TITLE
Move to new disk.

### DISCUSSION THREAD

**Discussion by ozgurerdogan** [psy0rz/zfs_autobackup#229] [General]
I need to move existing disk to a new disk by keeping all data along with snapshots. What is the best way for this. 
I am backing up Disk A > Disk B > Disk C
Disk B must be moved to new disk. I do not want to start full backup with new disk.

**Comment by psy0rz** [comment]
Cant you just temporary zpool attach the new disk to disk b? So it will be a mirror that syncs.

**Comment by ozgurerdogan** [comment]
You mean attach new disk and make it mirror with zfs? then detach old disk.?

**Reply by psy0rz** [reply]
correct. thats the easiest and safest method.

Dont make the mistake that you use `zpool add`, creating a striped situation that you cant undo!

**Comment by psy0rz** [comment]
still, you can use zfs-autobackup for this. just create a new pool, and point the target-path to it. then all neccesary snapshots )(according to thinning schedule of target) that are on disk A will be again transferred to B. 

After that you run zfs-autobackup again for B -> C


The thing is that zfs-autobackup doesnt keep any hidden state or anything like that, so you can just rerun with a new or empty target and it will do what you expect. :)

**Comment by ozgurerdogan** [comment]
_create a new pool, and point the target-path to it_
But then zfs-autobackup will creat full backup again. I need it to continue from where it left NOT from scracth.
So If I can make a exact copy of existing disk with "Correct way" then it should be fine. But I think (from by expericences), zfs-autobackup does not like anything done out of its control  :) and it throws "no common snapshot" issue so I have to let it do a full sync again.

**Comment by psy0rz** [comment]
Yes it will do a full sync, since the disk is still empty. :) If you do the zpool attach trick, its also basically the same amount of time and data, as that of a full sync.

You can do anything you like outside of zfs-autobackups control, as long as you make sure you have valid common snapshots. (or an empty target offcourse)

**Reply by psy0rz** [reply]
having said that: the next version of zfs-autobackup will have some improvements that make life easier when something does go wrong :)

**Reply by ozgurerdogan** [reply]
Looking forward to see next version...

**Reply by psy0rz** [reply]
btw, also look at the new page https://github.com/psy0rz/zfs_autobackup/wiki/Mounting if you sometimes have mysterious common snaphots issues.

**Comment by ozgurerdogan** [comment]
Is there any update on this ? I need to move a dataset to a new disk (On Proxmox). You know there is "zfs send -R" command which sends dataset with all snapshot. Can I use it and after sent delete snapshot that was create by zfs command only ?
I need to find a way to do this. Because time to time, I need to move dataset to new disk or node. And everytime, it creates full backup.

**Reply by psy0rz** [reply]
Well, if its a mirror use zpool attach.

Otherwise just use zfs send -R so it sends over all the sbapshots.
